src/Microsoft.DotNet.Cli.Utils/Command.cs

```diff
@@ -88,6 +98,43 @@ private static void ResolveExecutablePath(ref string executable, ref string args
                 args = $"/C \"{executable}\"";
                 executable = comSpec;
             }
+
+            return executable;
+        }
+
+        private static string ResolveExecutablePathFromProject(string executable, NuGetFramework framework)
+        {
+            if (framework == null) return null;
+
+            var projectRootPath = PathUtility.GetProjectRootPath();
+
+            if (projectRootPath == null) return null;
+
+            var commandName = Path.GetFileNameWithoutExtension(executable);
+
+            var projectContext = ProjectContext.Create(projectRootPath, framework);
+
+            var commandPackage = projectContext.LibraryManager.GetLibraries()
+                .Where(l => l.GetType() == typeof (PackageDescription))
+                .Select(l => l as PackageDescription)
+                .FirstOrDefault(p =>
+                {
+                    var fileNames = p.Library.Files
+                        .Select(Path.GetFileName)
+                        .Where(n => Path.GetFileNameWithoutExtension(n) == commandName)
+                        .ToList();
+
+                    return fileNames.Contains(commandName + FileNameSuffixes.ExeSuffix) &&
+                           fileNames.Contains(commandName + FileNameSuffixes.DynamicLib) &&
```

Comment:
I refactored our constants some more to make the intention more clear. This will be FileNameSuffixes.DotNet.DynamicLib == ".dll"